### ISSUE TITLE
unit ventilator object is incomplete in 1.11.0

### ISSUE THREAD

**Issue by MatthewSteen** [NREL/OpenStudio#2184]
The ZoneHVACUnitVentilator objects do not expose the fan component in the inspector.

https://unmethours.com/question/13699/supply-fan-name-type-in-for-an-unit-ventilator/

A FanConstantVolume object is added to a model for each unit ventilator, but according to the IO Ref this object should have all three fan type options.

http://bigladdersoftware.com/epx/docs/8-4/input-output-reference/group-zone-forced-air-units.html#field-supply-air-fan-object-type-1

**Comment by MatthewSteen** [comment]
Upon further inspection, the fan is created automagically but not connected to the parent object.

    OS:ZoneHVAC:UnitVentilator,
      {8bfd8b9f-0f07-4183-ba7e-5bbc5d0bdcc8}, !- Handle
      Unit Ventilator - Heating,              !- Name
      {9adb53f6-26f9-4483-aee1-914f1bc963b0}, !- Availability Schedule Name
      autosize,                               !- Maximum Supply Air Flow Rate {m3/s}
      VariablePercent,                        !- Outdoor Air Control Type
      autosize,                               !- Minimum Outdoor Air Flow Rate {m3/s}
      {d3f57cd1-3a1c-4bdc-9853-af033be26cb1}, !- Minimum Outdoor Air Schedule Name
      autosize,                               !- Maximum Outdoor Air Flow Rate {m3/s}
      {9adb53f6-26f9-4483-aee1-914f1bc963b0}, !- Maximum Outdoor Air Fraction or Temperature Schedule Name
      {a51d0596-bcf0-48e6-83a4-dbb46ba41c45}, !- Air Inlet Node Name
      {8edfa840-0342-415b-8135-fc4c9c644d61}, !- Air Outlet Node Name
      ,                                       !- Supply Air Fan Name
      ,                                       !- Supply Air Fan Operating Mode Schedule Name
      {23e1980a-ca87-4746-b705-69db61ec9486}, !- Heating Coil Name
      0.001,                                  !- Heating Convergence Tolerance
      ,                                       !- Cooling Coil Name
      0.001,                                  !- Cooling Convergence Tolerance
      ,                                       !- Availability Manager List Name
      ;                                       !- Design Specification ZoneHVAC Sizing Object Name

    OS:Fan:ConstantVolume,
      {c8571377-d663-4c01-aa01-ab498f03a3a6}, !- Handle
      Fan Constant Volume,                  !- Name
      {9adb53f6-26f9-4483-aee1-914f1bc963b0}, !- Availability Schedule Name
      ,                                       !- Fan Efficiency
      ,                                       !- Pressure Rise {Pa}
      AutoSize,                               !- Maximum Flow Rate {m3/s}
      ,                                       !- Motor Efficiency
      ,                                       !- Motor In Airstream Fraction
      ,                                       !- Air Inlet Node Name
      ,                                       !- Air Outlet Node Name
      ;                                       !- End-Use Subcategory

**Comment by mbadams5** [comment]
@MatthewSteen I don't see anything in the code that limits ZoneHVACUnitVentilator to only FanConstantVolume. In fact it looks like it can properly use all three (again, just from looking at the code only).

I will say that within hvac_library.osm file that the ZoneHVACUnitVentilator objects in there only use FanConstantVolume. What this means is that when you drag a ZoneHVACUnitVentilator from the library, it will generate that new object in your model, using a FanConstantVolume. To be able to use a ZoneHVACUnitVentilator object with another fan type, those options need to be added to hvac_library.osm or you need to use the API to change the attached Fan (quickest and easiest method).

At this point, no subcomponents (fans, coils, etc) of a parent object are exposed through the inspector.

**Comment by MatthewSteen** [comment]
Ok, so the lack of the fan component in the inspector is by design? 

In attempting to help the UnmetHours user name the UV fan, I tried using the API to get the supplyAirFan, but it causes an error.

The OSM block above was created by dropping a UV into a zone in a blank model. I tried all three of the UV object types and none of the CAV fans that were created for them were linked, i.e. the `Supply Air Fan` field is blank.
